### DISCUSSION TITLE
Support C++20 Module

### DISCUSSION THREAD

**Discussion by soroshsabz** [microsoft/vcpkg#17851] [Ideas]
ITNOA

I think the [C++20 Module](https://devblogs.microsoft.com/cppblog/standard-c20-modules-support-with-msvc-in-visual-studio-2019-version-16-8/) is the next era of [C++ dependency management](https://stackoverflow.com/q/57300495/1539100). So I think VCPKG need some good integration with it, and port all of its ports to support C++20 Module. and maybe `vcpkg` can play role like cargo for C++ so I think it is very important to fast decision and good design to make deep integration with C++20 Module.

The good news is [CMake 3.20 is support C++20 Module natively](https://gitlab.kitware.com/cmake/cmake/-/blob/master/Help/dev/experimental.rst) (also is experimental still now) and I think we can get help from that.

**Comment by soroshsabz** [comment]
related to #11177

**Comment by PhoebeHui** [comment]
@soroshsabz, thanks for bringing this up! we'd perfer to change this to discussion.

**Comment by mathisloge** [comment]
@soroshsabz unfortunately this is only a first step for cmake to pave the way for modules. There are a big number of missing bits to have a native integration of c++ modules. 
See the cmake issue https://gitlab.kitware.com/cmake/cmake/-/issues/18355

**Reply by soroshsabz** [reply]
@mathisloge you right, but because of merging https://gitlab.kitware.com/cmake/cmake/-/merge_requests/5562 , I think `vcpkg` can start working on it.

**Reply by mathisloge** [reply]
What are you expecting vcpkg to do regarding modules?
Libraries, cmake and compilers have to support modules.

For the installation vcpkg normally depends on what the build tools are installing

**Comment by kelteseth** [comment]
I have created https://arewemodulesyet.org/ to track modules usage. It would be nice if vcpkg itself would provide a list if a port supports modules
